src/Compilers/Core/Portable/Syntax/SyntaxNode.cs

```diff
@@ -748,6 +747,33 @@ public ChildSyntaxList ChildNodesAndTokens()
             return new ChildSyntaxList(this);
         }
 
+        internal ChildSyntaxList.Segment SiblingsAfterSelf()
+        {
+            int initialIndex = 0;
+            int terminalIndex = 0;
+            if (_parent != null)
+            {
+                initialIndex = _parent.GetIndexOfChild(this);
+                Debug.Assert(initialIndex >= 0, "Node couldn't be found in its parent.");
+                initialIndex &= int.MaxValue; // Doesn't affect non-negative numbers.
+                                              // Turns the not-found index (-1) into a virtually unreachable number, causing the segment to be empty.
+
+                terminalIndex = ChildSyntaxList.CountNodes(_parent.Green);
+            }
+            return new ChildSyntaxList.Segment(_parent!, initialIndex, terminalIndex);
```

Comment:
i don't understand the `!` on `_parent`.  The code above seems to show that having a null parent is fine.  So casting away null-ness seems explicitly wrong here.

Reply:
for a ChildSyntaxList.Segement, a null parent is only fine if initialIndex >= terminalIndex;
it's an edge-case where we should assert that this specific scenario is supported,
without losing nullability checks for all other scenarios

Reply:
i still don't get it.  is there a test we have that hits this case, so i can see when this would arise in real trees?

Reply:
`SyntaxNode._parent` in it of itself is nullable for root nodes and structured trivia which don't have a parent node,
so this returns an empty segment when SiblingsAfter/BeforeSelf is called on them as they don't have siblings.

Reply:
ok.  please write test that validates a use case where GetNext/PreviousToken on a token will end up hitting that case.  Thanks!

Reply:
And doc that as well.